workflows/preparegenome.nf

```diff
@@ -52,18 +56,39 @@ workflow {
     // Create a FAI genome index using samtools
     ch_fai = SAMTOOLS_FAIDX ( ch_fasta ).fai
 
+    FILTER_GTF ( params.gtf )
+
+    // iCount segment the raw annotation
+    RAW_ICOUNT_SEGMENT (
+        params.gtf,
+        ch_fai
+    )
 
-    // Create genome segmentation file using iCount
-    ICOUNT_SEGMENT (
+    // iCount segment the filtered annotation
+    FILTERED_ICOUNT_SEGMENT (
         FILTER_GTF.out.post_filtering_gtf,
         ch_fai
     )
 
+    RENAME (
+        FILTERED_ICOUNT_SEGMENT.out.regions,
+        "filtered.regions.gtf.gz"
+    )
+
+    // Resolve unannotated for iCount summary and other processes
+    RESOLVE_UNANNOTATED (
+        RAW_ICOUNT_SEGMENT.out.regions,      // filtered_segmentation
+        RENAME.out.renamed, // unfiltered_segmentation
+        FILTER_GTF.out.post_filtering_gtf,   // gtf
+        ch_fai,                              // fai
+    )
 
-    RESOLVE_UNANNOTATED(
-        ICOUNT_SEGMENT.out.regions, // segment
-        FILTER_GTF.out.post_filtering_gtf, // gtf
-        ch_fai, // fai
+    // Resolve unannotated for PEKA
+    RESOLVE_UNANNOTATED_GENIC_OTHER (
+        RAW_ICOUNT_SEGMENT.out.regions,      // filtered_segmentation
+        RENAME.out.renamed, // unfiltered_segmentation
+        FILTER_GTF.out.post_filtering_gtf,   // gtf
+        ch_fai,                              // fai
```

Comment:
Hi @marc-jones! The --genic_other flag should be passed here, is this is done with the extended argument?

Reply:
That's passed in [here](https://github.com/goodwright/imaps-nf/pull/7/files#diff-e8c87a50ddb1e50082721ad74cf232c842aa8ca6a01e62c7c9a60880fdf0a4adR38). It's a bit messy but I think (@samirelanduk ?) that's the best way to do it.

Reply:
Yeah haven't settled on a definitive way to pass params yet - this is absolutely fine for now.